extensions/wikia/ARecoveryEngine/ARecoveryModule.class.php

```diff
@@ -18,16 +15,31 @@ public function isPageFairRecoveryEnabled() {
 	/**
 	 * Checks whether SourcePoint recovery is enabled (on current wiki)
 	 *
-	 * $wgAdDriverEnableSourcePointRecovery === false; // disabled on wiki
-	 * $wgAdDriverEnableSourcePointRecovery === true; // enabled on wiki
+	 * @return bool
+	 */
+	public static function isSourcePointRecoveryEnabled() {
+		global $wgUser, $wgAdDriverEnableSourcePointRecovery;
+
+		return $wgUser->isAnon() && F::app()->checkSkin( [ 'oasis' ] ) && $wgAdDriverEnableSourcePointRecovery;
```

Comment:
Nit-picking: what do you think about having `isRecoveryEnabled()` and re-using it in `isPageFairRecoveryEnabled()` and `isSourcePointRecoveryEnabled()`? Pros: no code duplication, cons: worse readability?